In [1]:
# IMPORTING LIBRARIES

!pip install pytesseract
!pip install tesseract
!apt install tesseract-ocr
!apt install libtesseract-dev
!sudo apt install tesseract-ocr
!pip install Pillow
from PIL import Image
import pytesseract
import cv2
import sys
import os 

# MOUNTING THE GOOGLE DRIVE

from google.colab import drive
drive.mount('/content/gdrive')

  Created wheel for pytesseract: filename=pytesseract-0.3.7-py2.py3-none-any.whl size=13945 sha256=4b2b2bcbf3110b1554e6174ed7ff67c3f6f83480e8d286b07c10ef03063c01dd
  Stored in directory: /root/.cache/pip/wheels/81/20/7e/1dd0daad1575d5260916bb1e9781246430647adaef4b3ca3b3
Successfully built pytesseract
     |████████████████████████████████| 45.6MB 95kB/s 
  Created wheel for tesseract: filename=tesseract-0.1.3-cp37-none-any.whl size=45562571 sha256=7fb204e7be59a151660c2b0c8e9de040c3a25b2a9b6eaf8dea844fc3a5db9a52
  Stored in directory: /root/.cache/pip/wheels/82/1f/d9/24797b123379e4ea9511cf660835468b62dad609634cad2aba
Successfully built tesseract
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following package was automatically installed and is no longer required:
  libnvidia-common-460
Use 'apt autoremove' to remove it.
The following additional packages will be installed:
  tesseract-ocr-eng tesseract-ocr-osd
The following NEW package

NOTE: The result of all the techniques or proceedures applied are saved in respective folders for clarity and a more organised analysis.

In [2]:
# EXTRACTING REGION OF INTEREST

import cv2
import glob
from google.colab.patches import cv2_imshow

# PATH TO ORIGINAL IMAGES TO BE EXTRACTED
path = glob.glob("/content/gdrive/MyDrive/CA_3/images/*.jpg") 

# LOOPING THROUGH THE FILES IN THE FOLDER CONTAINING THE IMAGES TO BE EXTRACTED
img_number = 1
for file in path:
  print(file)
  img = cv2.imread(file)

  # CUSTOMIZING THE OPTIONS OF THE IMAGES
  custom_config = r'--oem 3 --psm 11'

  # SHAPE OF THE EXTRACTED IMAGES
  img = img[230:img.shape[0]-3000,5630:img.shape[1]-290]

  # SAVING THE EXTRACTED IMAGES INTO A SEPERATE FOLDER NAMED 'Extracted_images'
  cv2.imwrite('/content/gdrive/MyDrive/CA_3/Extracted_images/image'+str(img_number)+'.jpg', img)
  img_number +=1 

  # APPLYING RECTANGULAR BOUNDING BOXES ON THE ROI
  h, w, c = img.shape
  boxes = pytesseract.image_to_boxes(img)
  for b in boxes.splitlines():
    b = b.split(' ')
    img = cv2.rectangle(img, (int(b[1]), h - int(b[2])), (int(b[3]), h - int(b[4])), (255, 0, 0), 2) 
    
  # DISPLAYING THE IMAGES
  #cv2_imshow(img)
  cv2.waitKey(0)
  cv2.destroyAllWindows()

/content/gdrive/MyDrive/CA_3/images/RedactedIsometric1-1.jpg
/content/gdrive/MyDrive/CA_3/images/RedactedIsometric2-1.jpg
/content/gdrive/MyDrive/CA_3/images/RedactedIsometric3-1.jpg
/content/gdrive/MyDrive/CA_3/images/RedactedIsometric4-1.jpg
/content/gdrive/MyDrive/CA_3/images/RedactedIsometric5-1.jpg
/content/gdrive/MyDrive/CA_3/images/RedactedIsometric6-1.jpg
/content/gdrive/MyDrive/CA_3/images/RedactedIsometric7-1.jpg
/content/gdrive/MyDrive/CA_3/images/RedactedIsometric8-1.jpg
/content/gdrive/MyDrive/CA_3/images/RedactedIsometric9-1.jpg
/content/gdrive/MyDrive/CA_3/images/RedactedIsometric10-1.jpg
/content/gdrive/MyDrive/CA_3/images/RedactedIsometric11-1.jpg
/content/gdrive/MyDrive/CA_3/images/RedactedIsometric12-1.jpg
/content/gdrive/MyDrive/CA_3/images/RedactedIsometric13-1.jpg
/content/gdrive/MyDrive/CA_3/images/RedactedIsometric14-1.jpg
/content/gdrive/MyDrive/CA_3/images/RedactedIsometric16-1.jpg
/content/gdrive/MyDrive/CA_3/images/RedactedIsometric15-1.jpg
/content/gdrive/M

In [3]:

# APPLYING GRAYSCALE AND BINARISATION TECHNIQUE ON THE EXTRACTED REGION OF INTEREST

from google.colab.patches import cv2_imshow

# PATH TO THE EXTRACTED ROI
path = glob.glob("/content/gdrive/MyDrive/CA_3/Extracted_images/*.jpg")

# LOOPING THROUGH FILES IN THE EXTRACTED ROI FOLDER
for file in path:
  print(file)
  img = cv2.imread(file) # READING THE FILES

  # GRAYSCALE FUNCTION
  grayscale_img = cv2.cvtColor(img,cv2.COLOR_BGR2GRAY)
  img = cv2.adaptiveThreshold(grayscale_img,255,cv2.ADAPTIVE_THRESH_GAUSSIAN_C,cv2.THRESH_BINARY,11,2)

  # SAVING THE GRAYSCALE IMAGES INSIDE A SEPERATE FOLDER NAMED 'Grayscale_images'
  cv2.imwrite('/content/gdrive/MyDrive/CA_3/Grayscale_images/images'+str(img_number)+'.jpg', img)
  img_number +=1

  # APPLYING RECTANGULAR BOUNDING BOXES ON THE EXTRACTED GRAYSCALE IMAGES
  h, w = img.shape
  boxes = pytesseract.image_to_boxes(img)
  for b in boxes.splitlines():
    b = b.split(' ')
    img = cv2.rectangle(img, (int(b[1]), h - int(b[2])), (int(b[3]), h - int(b[4])), (255, 0, 0), 2) 
  
  # IMAGE VISUALIZATION
  #cv2_imshow(img)
  cv2.waitKey(0)
  cv2.destroyAllWindows()


/content/gdrive/MyDrive/CA_3/Extracted_images/image1.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image2.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image3.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image4.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image5.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image6.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image7.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image8.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image9.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image10.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image11.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image12.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image13.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image14.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image15.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image16.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image17.jpg
/content/gdrive/MyDrive

In [4]:
# PERFORMING EROSION ON THE EXTRACTED ROI

import cv2
import numpy as np
import glob
from google.colab.patches import cv2_imshow

# PATH TO THE ROI IMAGES
path = glob.glob("/content/gdrive/MyDrive/CA_3/Extracted_images/*.jpg")

# LOOPING THROUGH THE IMAGES IN THE ROI FOLDER
for file in path:
  print(file)
  img = cv2.imread(file)

  # CUSTOMIZING OPTIONS 
  custom_config = r'--oem 3 --psm 11'
  
  # EROSION FUNCTION
  kernel = np.ones((3,3),np.uint8)
  erosion_img = cv2.erode(img,kernel,iterations = 5)

  # SAVING THE EROSION IMAGES INTO A SEPERATE FOLDER NAMED 'Erosion'
  cv2.imwrite('/content/gdrive/MyDrive/CA_3/Erosion/images'+str(img_number)+'.jpg', img)
  img_number +=1

  # APPLYING RECTANGULAR BOUNDING BOXES ON THE EROSION IMAGES
  h, w, c = img.shape
  boxes = pytesseract.image_to_boxes(img)
  for b in boxes.splitlines():
    b = b.split(' ')
    img = cv2.rectangle(img, (int(b[1]), h - int(b[2])), (int(b[3]), h - int(b[4])), (255, 0, 0), 2) #OpenCV is in Blue, Green, Red format.
  
  # DISPLAYING THE IMAGES
  #cv2_imshow(img)
  cv2.waitKey(0)
  cv2.destroyAllWindows()

/content/gdrive/MyDrive/CA_3/Extracted_images/image1.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image2.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image3.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image4.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image5.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image6.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image7.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image8.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image9.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image10.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image11.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image12.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image13.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image14.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image15.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image16.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image17.jpg
/content/gdrive/MyDrive

In [5]:
# CONTOUR DETECTION OF THE ROI IMAGES

from PIL import Image
import pytesseract
import cv2
import glob
from google.colab.patches import cv2_imshow

# PATH TO THE FOLDER CONTAINING THE ROI IMAGES
path = glob.glob("/content/gdrive/MyDrive/CA_3/Extracted_images/*.jpg")

# LOOPING THROUGH THE IMAGES IN THE ROI FOLDER
for file in path:
  print(file)
  img = cv2.imread(file)
  
  # FUNCTION FOR CONTOUR DETECTION ON THE ROI IMAGES
  img = cv2.cvtColor(img, cv2.COLOR_BGR2GRAY) # CONVERTING THE IMAGE COLOUR TO GRAY  
  contours = cv2.findContours(img, cv2.RETR_TREE, cv2.CHAIN_APPROX_SIMPLE) # FINDING CONTOURS
  cv2.drawContours(img,[],-1,(255,255,255),-1) # DRAWING THE CONTOURS
  ret, img = cv2.threshold(img, 150, 255, cv2.THRESH_BINARY) # Applying binary thresholding

  # SAVING THE EROSION IMAGES INTO A SEPERATE FOLDER NAMED 'Contour'
  cv2.imwrite('/content/gdrive/MyDrive/CA_3/Contour/image'+str(img_number)+'.jpg', img)
  img_number +=1 

  # APPLYING RECTANGULAR BOUNDING BOXES ON THE EROSION IMAGES
  boxes = pytesseract.image_to_boxes(img)
  for b in boxes.splitlines():
    b = b.split(' ')
    img = cv2.rectangle(img, (int(b[1]), h - int(b[2])), (int(b[3]), h - int(b[4])), (255, 0, 0), 2) #OpenCV is in Blue, Green, Red format.
  
  # DISPLAYING THE IMAGES
  #cv2_imshow(img)
  cv2.waitKey(0)
  cv2.destroyAllWindows()



/content/gdrive/MyDrive/CA_3/Extracted_images/image1.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image2.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image3.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image4.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image5.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image6.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image7.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image8.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image9.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image10.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image11.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image12.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image13.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image14.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image15.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image16.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image17.jpg
/content/gdrive/MyDrive

In [6]:
# AVERAGE BLURING

from PIL import Image
import pytesseract
import cv2
from PIL import Image, ImageFilter
import glob
from google.colab.patches import cv2_imshow

# PATH TO THE FOLDER CONTAINING THE ROI IMAGES
path = glob.glob("/content/gdrive/MyDrive/CA_3/Extracted_images/*.jpg")

# LOOPING THROUGH THE IMAGES IN THE ROI FOLDER
for file in path:
  print(file)
  img = cv2.imread(file)
  
  # FUNCTION FOR IMAGE AVERAGE BLURING
  img = cv2.blur(img,(10,10)) 
  
  # SAVING THE EROSION IMAGES INTO A SEPERATE FOLDER NAMED 'Blur_images'
  cv2.imwrite('/content/gdrive/MyDrive/CA_3/Blur_images/image'+str(img_number)+'.jpg', img)
  img_number +=1 

  # APPLYING RECTANGULAR BOUNDING BOXES ON THE EROSION IMAGES
  boxes = pytesseract.image_to_boxes(img)
  for b in boxes.splitlines():
    b = b.split(' ')
    img = cv2.rectangle(img, (int(b[1]), h - int(b[2])), (int(b[3]), h - int(b[4])), (255, 0, 0), 2) #OpenCV is in Blue, Green, Red format.
  #cv2_imshow(img)
  cv2.waitKey(0)
  cv2.destroyAllWindows()




/content/gdrive/MyDrive/CA_3/Extracted_images/image1.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image2.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image3.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image4.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image5.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image6.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image7.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image8.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image9.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image10.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image11.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image12.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image13.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image14.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image15.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image16.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image17.jpg
/content/gdrive/MyDrive

In [7]:
# GAUSSIAN BLURING TECHNIQUE

# IMPORTING AND LOADING LIBRARIES 
from PIL import Image
import pytesseract
import cv2
from PIL import Image, ImageFilter
import glob
from google.colab.patches import cv2_imshow

# # PATH TO THE FOLDER CONTAINING THE ROI IMAGES
path = glob.glob("/content/gdrive/MyDrive/CA_3/Extracted_images/*.jpg")

# LOOPING THROUGH THE IMAGES IN THE ROI FOLDER
for file in path:
  print(file)
  img = cv2.imread(file)
  
  # FUNCTION FOR IMAGE GAUSSIAN BLURING
  gausBlur = cv2.GaussianBlur(img, (5,5),0)  
  img = cv2.blur(img,(10,10)) 

  # SAVING THE EROSION IMAGES INTO A SEPERATE FOLDER NAMED 'Gaussian_Blur'
  cv2.imwrite('/content/gdrive/MyDrive/CA_3/GaussianBlur_images/image'+str(img_number)+'.jpg', img)
  img_number +=1 

  # APPLYING RECTANGULAR BOUNDING BOXES ON THE EROSION IMAGES
  boxes = pytesseract.image_to_boxes(img)
  for b in boxes.splitlines():
    b = b.split(' ')
    img = cv2.rectangle(img, (int(b[1]), h - int(b[2])), (int(b[3]), h - int(b[4])), (255, 0, 0), 2) #OpenCV is in Blue, Green, Red format.
  
  # VISUALIZING THE IMAGES
  #cv2_imshow(gausBlur)
  cv2.waitKey(0)
  cv2.destroyAllWindows()




/content/gdrive/MyDrive/CA_3/Extracted_images/image1.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image2.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image3.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image4.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image5.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image6.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image7.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image8.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image9.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image10.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image11.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image12.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image13.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image14.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image15.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image16.jpg
/content/gdrive/MyDrive/CA_3/Extracted_images/image17.jpg
/content/gdrive/MyDrive

TEXT EXTRACTION

In [9]:
# EXTRACTING THE TEXT IN THE ROI IMAGES 

import os # IMPORTING OS LIBRARY

# LOOPING THROUGH THE ROI IMAGES
list_of_images = [] 
for root, dir, files in os.walk('/content/gdrive/MyDrive/Images_to_extract/Extracted_images2/'):
  for names in files:
    list_of_images.append("/content/gdrive/MyDrive/Images_to_extract/Extracted_images2/" + names)

# CONVERTING THE IMAGES TO STRINGS
text_list = []
for image in list_of_images:
  print(image)
  text = str(((pytesseract.image_to_string(Image.open(image), lang='eng')))) 
  text = text.replace('-\n', '')
  text_list.append(text)
 
# SAVING THE EXTRACTED TEXT INTO A SEPERATE FOLDER NAMED 'Text'
for i in range(len(text_list)):
  with open ("/content/gdrive/MyDrive/CA_3/Text/"+list_of_images[i][-11:]+".txt", 'w') as f:
    f.write(text_list[i])
    print(i)


/content/gdrive/MyDrive/Images_to_extract/Extracted_images2/image2.jpg
/content/gdrive/MyDrive/Images_to_extract/Extracted_images2/image1.jpg
/content/gdrive/MyDrive/Images_to_extract/Extracted_images2/image3.jpg
/content/gdrive/MyDrive/Images_to_extract/Extracted_images2/image4.jpg
/content/gdrive/MyDrive/Images_to_extract/Extracted_images2/image5.jpg
/content/gdrive/MyDrive/Images_to_extract/Extracted_images2/image6.jpg
/content/gdrive/MyDrive/Images_to_extract/Extracted_images2/image7.jpg
/content/gdrive/MyDrive/Images_to_extract/Extracted_images2/image8.jpg
/content/gdrive/MyDrive/Images_to_extract/Extracted_images2/image9.jpg
/content/gdrive/MyDrive/Images_to_extract/Extracted_images2/image10.jpg
/content/gdrive/MyDrive/Images_to_extract/Extracted_images2/image11.jpg
/content/gdrive/MyDrive/Images_to_extract/Extracted_images2/image12.jpg
/content/gdrive/MyDrive/Images_to_extract/Extracted_images2/image13.jpg
/content/gdrive/MyDrive/Images_to_extract/Extracted_images2/image14.jpg
/

In [10]:
# EXTRACTING THE TEXT FROM EROSION IMAGES 

import os

# LOOPING THROUGH THE ROI IMAGES
list_of_image = [] 
for root, dir, files in os.walk('/content/gdrive/MyDrive/CA_3/Erosion/'):
  for names in files:
    list_of_image.append("/content/gdrive/MyDrive/CA_3/Erosion/" + names)

# CONVERTING THE EROSION IMAGES TO STRINGS
text_list = [] 
for image in list_of_image:
  print(image)
  text = str(((pytesseract.image_to_string(Image.open(image), lang='eng')))) 
  text = text.replace('-\n', '')
  text_list.append(text)

# SAVING THE EXTRACTED TEXT INTO A SEPERATE FOLDER NAMED Erosion_text
for i in range(len(text_list)):
  with open ("/content/gdrive/MyDrive/CA_3/Erosion_text/"+list_of_image[i][-11:]+".txt", 'w') as f:
    f.write(text_list[i])
    print(i)


/content/gdrive/MyDrive/CA_3/Erosion/images101.jpg
/content/gdrive/MyDrive/CA_3/Erosion/images102.jpg
/content/gdrive/MyDrive/CA_3/Erosion/images103.jpg
/content/gdrive/MyDrive/CA_3/Erosion/images104.jpg
/content/gdrive/MyDrive/CA_3/Erosion/images105.jpg
/content/gdrive/MyDrive/CA_3/Erosion/images106.jpg
/content/gdrive/MyDrive/CA_3/Erosion/images107.jpg
/content/gdrive/MyDrive/CA_3/Erosion/images108.jpg
/content/gdrive/MyDrive/CA_3/Erosion/images109.jpg
/content/gdrive/MyDrive/CA_3/Erosion/images110.jpg
/content/gdrive/MyDrive/CA_3/Erosion/images111.jpg
/content/gdrive/MyDrive/CA_3/Erosion/images112.jpg
/content/gdrive/MyDrive/CA_3/Erosion/images113.jpg
/content/gdrive/MyDrive/CA_3/Erosion/images114.jpg
/content/gdrive/MyDrive/CA_3/Erosion/images115.jpg
/content/gdrive/MyDrive/CA_3/Erosion/images116.jpg
/content/gdrive/MyDrive/CA_3/Erosion/images117.jpg
/content/gdrive/MyDrive/CA_3/Erosion/images118.jpg
/content/gdrive/MyDrive/CA_3/Erosion/images119.jpg
/content/gdrive/MyDrive/CA_3/Er

In [11]:
# EXTRACTING THE TEXT FROM GRAYSCALE IMAGES 

import os

# LOOPING THROUGH THE GRAYSCALE IMAGES
list_of_image = [] 
for root, dir, files in os.walk('/content/gdrive/MyDrive/CA_3/Grayscale_images'):
  for names in files:
    list_of_image.append("/content/gdrive/MyDrive/CA_3/Grayscale_images/" + names)

# CONVERTING THE GRAYSCALE IMAGES TO STRINGS
text_list = [] 
for image in list_of_image:
  print(image)
  text = str(((pytesseract.image_to_string(Image.open(image), lang='eng')))) 
  text = text.replace('-\n', '')
  text_list.append(text)

# SAVING AND PRINTING THE EXTRACTED TEXT INTO A SEPERATE FOLDER NAMED 'Grayscale_text'
for i in range(len(text_list)):
  with open ("/content/gdrive/MyDrive/CA_3/Grayscale_text/"+list_of_image[i][-11:]+".txt", 'w') as f:
    f.write(text_list[i])
    print(i)


/content/gdrive/MyDrive/CA_3/Grayscale_images/images51.jpg
/content/gdrive/MyDrive/CA_3/Grayscale_images/images52.jpg
/content/gdrive/MyDrive/CA_3/Grayscale_images/images53.jpg
/content/gdrive/MyDrive/CA_3/Grayscale_images/images54.jpg
/content/gdrive/MyDrive/CA_3/Grayscale_images/images55.jpg
/content/gdrive/MyDrive/CA_3/Grayscale_images/images56.jpg
/content/gdrive/MyDrive/CA_3/Grayscale_images/images57.jpg
/content/gdrive/MyDrive/CA_3/Grayscale_images/images58.jpg
/content/gdrive/MyDrive/CA_3/Grayscale_images/images59.jpg
/content/gdrive/MyDrive/CA_3/Grayscale_images/images60.jpg
/content/gdrive/MyDrive/CA_3/Grayscale_images/images61.jpg
/content/gdrive/MyDrive/CA_3/Grayscale_images/images62.jpg
/content/gdrive/MyDrive/CA_3/Grayscale_images/images63.jpg
/content/gdrive/MyDrive/CA_3/Grayscale_images/images64.jpg
/content/gdrive/MyDrive/CA_3/Grayscale_images/images65.jpg
/content/gdrive/MyDrive/CA_3/Grayscale_images/images66.jpg
/content/gdrive/MyDrive/CA_3/Grayscale_images/images67.j

In [12]:
# EXTRACTING THE TEXT IN THE CONTOUR IMAGES 

import os

# LOOPING THROUGH THE CONTOUR IMAGES IN THE CONTOUR FOLDER
list_of_images = [] 
for root, dir, files in os.walk('/content/gdrive/MyDrive/CA_3/Contour/'):
  for names in files:
    list_of_images.append("/content/gdrive/MyDrive/CA_3/Contour/" + names)

# CONVERTING THE CONTOUR IMAGES TO STRINGS
text_list = [] 
for image in list_of_images:
  print(image)
  text = str(((pytesseract.image_to_string(Image.open(image), lang='eng')))) 
  text = text.replace('-\n', '')
  text_list.append(text)

# SAVING AND PRINTING THE EXTRACTED TEXT INTO A SEPERATE FOLDER NAMED 'Contour_text'
for i in range(len(text_list)):
  with open ("/content/gdrive/MyDrive/CA_3/Contour_text/"+list_of_images[i][-11:]+".txt", 'w') as f:
    f.write(text_list[i])
    #print(i)


/content/gdrive/MyDrive/CA_3/Contour/image151.jpg
/content/gdrive/MyDrive/CA_3/Contour/image152.jpg
/content/gdrive/MyDrive/CA_3/Contour/image153.jpg
/content/gdrive/MyDrive/CA_3/Contour/image154.jpg
/content/gdrive/MyDrive/CA_3/Contour/image155.jpg
/content/gdrive/MyDrive/CA_3/Contour/image156.jpg
/content/gdrive/MyDrive/CA_3/Contour/image157.jpg
/content/gdrive/MyDrive/CA_3/Contour/image158.jpg
/content/gdrive/MyDrive/CA_3/Contour/image159.jpg
/content/gdrive/MyDrive/CA_3/Contour/image160.jpg
/content/gdrive/MyDrive/CA_3/Contour/image161.jpg
/content/gdrive/MyDrive/CA_3/Contour/image162.jpg
/content/gdrive/MyDrive/CA_3/Contour/image163.jpg
/content/gdrive/MyDrive/CA_3/Contour/image164.jpg
/content/gdrive/MyDrive/CA_3/Contour/image165.jpg
/content/gdrive/MyDrive/CA_3/Contour/image166.jpg
/content/gdrive/MyDrive/CA_3/Contour/image167.jpg
/content/gdrive/MyDrive/CA_3/Contour/image168.jpg
/content/gdrive/MyDrive/CA_3/Contour/image169.jpg
/content/gdrive/MyDrive/CA_3/Contour/image170.jpg


In [13]:
# EXTRACTING THE TEXT FROM THE AVERAGE BLUR IMAGES 

import os

# LOOPING THROUGH THE AVERAGE BLUR IMAGES IN THE AVERAGE BLUR FOLDER
list_of_image = [] 
for root, dir, files in os.walk('/content/gdrive/MyDrive/CA_3/Blur_images'):
  for names in files:
    list_of_image.append("/content/gdrive/MyDrive/CA_3/Blur_images/" + names)

# CONVERTING THE AVERAGE BLUR IMAGES TO STRINGS
text_list = [] 
for image in list_of_image:
  print(image)
  text = str(((pytesseract.image_to_string(Image.open(image), lang='eng')))) 
  text = text.replace('-\n', '')
  text_list.append(text)

# SAVING AND PRINTING THE EXTRACTED TEXT INTO A SEPERATE FOLDER NAMED 'AverageBlur_text'
for i in range(len(text_list)):
  with open ("/content/gdrive/MyDrive/CA_3/Blur_text"+list_of_image[i][-11:]+".txt", 'w') as f:
    f.write(text_list[i])
    print(i)


/content/gdrive/MyDrive/CA_3/Blur_images/image201.jpg
/content/gdrive/MyDrive/CA_3/Blur_images/image202.jpg
/content/gdrive/MyDrive/CA_3/Blur_images/image203.jpg
/content/gdrive/MyDrive/CA_3/Blur_images/image204.jpg
/content/gdrive/MyDrive/CA_3/Blur_images/image205.jpg
/content/gdrive/MyDrive/CA_3/Blur_images/image206.jpg
/content/gdrive/MyDrive/CA_3/Blur_images/image207.jpg
/content/gdrive/MyDrive/CA_3/Blur_images/image208.jpg
/content/gdrive/MyDrive/CA_3/Blur_images/image209.jpg
/content/gdrive/MyDrive/CA_3/Blur_images/image210.jpg
/content/gdrive/MyDrive/CA_3/Blur_images/image211.jpg
/content/gdrive/MyDrive/CA_3/Blur_images/image212.jpg
/content/gdrive/MyDrive/CA_3/Blur_images/image213.jpg
/content/gdrive/MyDrive/CA_3/Blur_images/image214.jpg
/content/gdrive/MyDrive/CA_3/Blur_images/image215.jpg
/content/gdrive/MyDrive/CA_3/Blur_images/image216.jpg
/content/gdrive/MyDrive/CA_3/Blur_images/image217.jpg
/content/gdrive/MyDrive/CA_3/Blur_images/image218.jpg
/content/gdrive/MyDrive/CA_3

In [14]:
# EXTRACTING THE TEXT FROM THE GAUSSIAN BLUR IMAGES 

import os

# LOOPING THROUGH THE GAUSSIAN BLUR IMAGES IN THE GAUSSIAN BLUR FOLDER
list_of_image = [] 
for root, dir, files in os.walk('/content/gdrive/MyDrive/CA_3/GaussianBlur_images'):
  for names in files:
    list_of_image.append("/content/gdrive/MyDrive/CA_3/GaussianBlur_images/" + names)

# CONVERTING THE GAUSSIAN BLUR IMAGES TO STRINGS
text_list = [] 
for image in list_of_image:
  print(image)
  text = str(((pytesseract.image_to_string(Image.open(image), lang='eng')))) 
  text = text.replace('-\n', '')
  text_list.append(text)

# SAVING AND PRINTING THE EXTRACTED TEXT INTO A SEPERATE FOLDER NAMED 'GaussianBlur_text'
for i in range(len(text_list)):
  with open ("/content/gdrive/MyDrive/CA_3/GaussianBlur_text"+list_of_image[i][-11:]+".txt", 'w') as f:
    f.write(text_list[i])
    print(i)


/content/gdrive/MyDrive/CA_3/GaussianBlur_images/image251.jpg
/content/gdrive/MyDrive/CA_3/GaussianBlur_images/image252.jpg
/content/gdrive/MyDrive/CA_3/GaussianBlur_images/image253.jpg
/content/gdrive/MyDrive/CA_3/GaussianBlur_images/image254.jpg
/content/gdrive/MyDrive/CA_3/GaussianBlur_images/image255.jpg
/content/gdrive/MyDrive/CA_3/GaussianBlur_images/image256.jpg
/content/gdrive/MyDrive/CA_3/GaussianBlur_images/image257.jpg
/content/gdrive/MyDrive/CA_3/GaussianBlur_images/image258.jpg
/content/gdrive/MyDrive/CA_3/GaussianBlur_images/image259.jpg
/content/gdrive/MyDrive/CA_3/GaussianBlur_images/image260.jpg
/content/gdrive/MyDrive/CA_3/GaussianBlur_images/image261.jpg
/content/gdrive/MyDrive/CA_3/GaussianBlur_images/image262.jpg
/content/gdrive/MyDrive/CA_3/GaussianBlur_images/image263.jpg
/content/gdrive/MyDrive/CA_3/GaussianBlur_images/image264.jpg
/content/gdrive/MyDrive/CA_3/GaussianBlur_images/image265.jpg
/content/gdrive/MyDrive/CA_3/GaussianBlur_images/image266.jpg
/content

USING REGULAR EXPRESSION TO SEARCH FOR STRINGS

In [15]:
# USING REGULAR EXPRESSION TO SEARCH FOR STRINGS IN THE GROUND TRUTH TEXT FILES

# IMPORTING LIBRARIES
import os
import re
from PIL import Image
import pytesseract
import cv2
import glob
from google.colab.patches import cv2_imshow

# PATH TO THE GROUND TRUTH TEXT FILES
path = glob.glob("//content/gdrive/MyDrive/CA_3/ground_truths/*.txt")

# LOOPING THROUGH THE GROUND TRUTH FILES
for path in os.listdir('/content/gdrive/MyDrive/CA_3/ground_truths/'):
  with open('/content/gdrive/MyDrive/CA_3/ground_truths/'+ path) as f:
    fin = f.read()
    try:
      print(re.search('GALV',fin)) # SEARCHING FOR THE WORD 'GALV' IN THE GROUND TRUTH TEXT FILES
      print(path)
    except:
      print('Not Matched')
  print('--------')



<re.Match object; span=(483, 487), match='GALV'>
GroundTruth-BOM21.txt
--------
<re.Match object; span=(435, 439), match='GALV'>
GroundTruth-BOM22.txt
--------
<re.Match object; span=(914, 918), match='GALV'>
GroundTruth-BOM23.txt
--------
<re.Match object; span=(767, 771), match='GALV'>
GroundTruth-BOM24.txt
--------
<re.Match object; span=(860, 864), match='GALV'>
GroundTruth-BOM25.txt
--------
<re.Match object; span=(835, 839), match='GALV'>
GroundTruth-BOM26.txt
--------
<re.Match object; span=(437, 441), match='GALV'>
GroundTruth-BOM27.txt
--------
<re.Match object; span=(482, 486), match='GALV'>
GroundTruth-BOM28.txt
--------
<re.Match object; span=(386, 390), match='GALV'>
GroundTruth-BOM29.txt
--------
<re.Match object; span=(433, 437), match='GALV'>
GroundTruth-BOM30.txt
--------
None
GroundTruth-BOM10.txt
--------
None
GroundTruth-BOM6.txt
--------
None
GroundTruth-BOM7.txt
--------
None
GroundTruth-BOM4.txt
--------
None
GroundTruth-BOM3.txt
--------
None
GroundTruth-BOM8.tx

COUNTING THE NUMBER OF WORDS IN TEXT FILES

In [16]:
# COUNTING THE NUMBER OF WORDS IN EACH GROUND TRUTH TEXT FILE

# IMPORTING LIBRARIES
import os
import glob
from google.colab.patches import cv2_imshow

# PATH TO THE GROUND TRUTH TEXT FILES
path = glob.glob("/content/gdrive/MyDrive/CA_3/ground_truths/*.txt")

# LOOPING THROUGH THE TEXT FILES
for path in os.listdir('/content/gdrive/MyDrive/CA_3/ground_truths/'):
  with open('/content/gdrive/MyDrive/CA_3/ground_truths/'+ path) as f:
    fin = f.read()
    words = fin.split()
    print('Number of words in text file :', len(words)) # COUNTING THE WORDS IN EACH FILE
    print(path)


Number of words in text file : 238
GroundTruth-BOM21.txt
Number of words in text file : 214
GroundTruth-BOM22.txt
Number of words in text file : 325
GroundTruth-BOM23.txt
Number of words in text file : 321
GroundTruth-BOM24.txt
Number of words in text file : 347
GroundTruth-BOM25.txt
Number of words in text file : 336
GroundTruth-BOM26.txt
Number of words in text file : 217
GroundTruth-BOM27.txt
Number of words in text file : 271
GroundTruth-BOM28.txt
Number of words in text file : 183
GroundTruth-BOM29.txt
Number of words in text file : 245
GroundTruth-BOM30.txt
Number of words in text file : 157
GroundTruth-BOM10.txt
Number of words in text file : 157
GroundTruth-BOM6.txt
Number of words in text file : 157
GroundTruth-BOM7.txt
Number of words in text file : 129
GroundTruth-BOM4.txt
Number of words in text file : 142
GroundTruth-BOM3.txt
Number of words in text file : 157
GroundTruth-BOM8.txt
Number of words in text file : 126
GroundTruth-BOM1.txt
Number of words in text file : 157
Gr

In [17]:
# COUNTING THE NUMBER OF WORDS IN EACH TEXT DOCUMENT OF THE ORIGINAL IMAGES

import os
import glob
from google.colab.patches import cv2_imshow

# PATH TO THE TEXT FILES OF THE ORIGINAL IMAGES
path = glob.glob("/content/gdrive/MyDrive/new_folder/*.txt")

# LOOPING THROUGH THE TEXT FILES
for path in os.listdir('/content/gdrive/MyDrive/new_folder/'):
  with open('/content/gdrive/MyDrive/new_folder/'+ path) as f:
    fin = f.read()
    words = fin.split()
    print('Number of words in text file :', len(words)) # PRINTING NUMBER OF WORDS
    print(path)


Number of words in text file : 245
time_RedactedIsometric1-1.jpg.txt
Number of words in text file : 133
time_RedactedIsometric2-1.jpg.txt
Number of words in text file : 5166
time_RedactedIsometric3-1.jpg.txt
Number of words in text file : 240
time_RedactedIsometric4-1.jpg.txt
Number of words in text file : 663
time_RedactedIsometric5-1.jpg.txt
Number of words in text file : 618
time_RedactedIsometric6-1.jpg.txt
Number of words in text file : 739
time_RedactedIsometric7-1.jpg.txt
Number of words in text file : 745
time_RedactedIsometric8-1.jpg.txt
Number of words in text file : 686
time_RedactedIsometric9-1.jpg.txt
Number of words in text file : 675
time_RedactedIsometric10-1.jpg.txt
Number of words in text file : 787
time_RedactedIsometric11-1.jpg.txt
Number of words in text file : 611
time_RedactedIsometric12-1.jpg.txt
Number of words in text file : 704
time_RedactedIsometric13-1.jpg.txt
Number of words in text file : 669
time_RedactedIsometric14-1.jpg.txt
Number of words in text file

In [19]:
# COUNTING THE NUMBER OF WORDS IN EACH TEXT DOCUMENT OF THE ROI IMAGES

import os
import glob
from google.colab.patches import cv2_imshow

# PATH TO THE TEXT FILES OF THE ROI IMAGES
path = glob.glob("/content/gdrive/MyDrive/CA_3/Text/*.txt")

# LOOPING THROUGH THE ROI TEXT FILES
for path in os.listdir('/content/gdrive/MyDrive/CA_3/Text/'):
  with open('/content/gdrive/MyDrive/CA_3/Text/'+ path) as f:
    fin = f.read()
    words = fin.split()
    print('Number of words in text file :', len(words)) # PRINTING NUMBER OF WORDS
    print(path) # PRINTING EACH FILE PATH


Number of words in text file : 22
image2.jpg.txt
Number of words in text file : 160
image1.jpg.txt
Number of words in text file : 114
image3.jpg.txt
Number of words in text file : 59
image4.jpg.txt
Number of words in text file : 226
image5.jpg.txt
Number of words in text file : 184
image6.jpg.txt
Number of words in text file : 200
image7.jpg.txt
Number of words in text file : 241
image8.jpg.txt
Number of words in text file : 193
image9.jpg.txt
Number of words in text file : 197
image10.jpg.txt
Number of words in text file : 220
image11.jpg.txt
Number of words in text file : 192
image12.jpg.txt
Number of words in text file : 224
image13.jpg.txt
Number of words in text file : 224
image14.jpg.txt
Number of words in text file : 83
image15.jpg.txt
Number of words in text file : 157
image16.jpg.txt
Number of words in text file : 257
image17.jpg.txt
Number of words in text file : 295
image18.jpg.txt
Number of words in text file : 223
image19.jpg.txt
Number of words in text file : 216
image20.

CONCLUSIVELY: the number of words in each of the extracted ROI text files is smaller when comapared to those of the Ground Truths and text files from original images. 

**EVALUATION/ACCURACY VALUES**

In [20]:
# GROUND TRUTHS VS TEXT FROM ORIGINAL IMAGES

# IMPORTING LIBRARY
from difflib import SequenceMatcher

# PATH TO THE GROUND TRUTH TEXT FILES
path1 = glob.glob("/content/gdrive/MyDrive/CA_3/ground_truths/*.txt")

# LOOPING THROUGH THE FOLDER CONTAINING THE GROUND TRUTH TEXT FILES
for path1 in os.listdir('/content/gdrive/MyDrive/CA_3/ground_truths/'):
  with open('/content/gdrive/MyDrive/CA_3/ground_truths/'+ path1) as f1:
    fin1 = f1.read()

# PATH TO TEXT FILES FROM THE ORIGINAL IMAGES
path2 = glob.glob("/content/gdrive/MyDrive/new_folder/*.txt")

# LOOPING THROUGH THE FOLDER CONTAINING THE TEXT FILES OF THE ORIGINAL IMAGES
for path2 in os.listdir('/content/gdrive/MyDrive/new_folder/'):
  with open('/content/gdrive/MyDrive/new_folder/'+ path2) as f2:
    fin2 = f2.read()

# DETERMING THE ACCURACY VALUE OF THE TEXT FILES IN THE TWO FOLDERS
seq = SequenceMatcher(a=fin1, b=fin2)
print(seq.ratio())

# ACCURACY VALUE = 0.2485207100591716

0.2485207100591716


In [22]:
# GROUND TRUTHS VS TEXT FROM ROI IMAGES

from difflib import SequenceMatcher

# PATH TO THE GROUND TRUTH TEXT FILES
path1 = glob.glob("/content/gdrive/MyDrive/CA_3/ground_truths/*.txt")

# LOOPING THROUGH THE FOLDER CONTAINING THE GROUND TRUTH TEXT FILES
for path1 in os.listdir('/content/gdrive/MyDrive/CA_3/ground_truths/'):
  with open('/content/gdrive/MyDrive/CA_3/ground_truths/'+ path1) as f1:
    fin1 = f1.read()

# PATH TO TEXT FILES FROM THE ORIGINAL IMAGES
path2 = glob.glob("/content/gdrive/MyDrive/CA_3/Text/*.txt")

# LOOPING THROUGH THE FOLDER CONTAINING THE ROI TEXT FILES
for path2 in os.listdir('/content/gdrive/MyDrive/CA_3/Text/'):
  with open('/content/gdrive/MyDrive/CA_3/Text/'+ path2) as f2:
    fin2 = f2.read()

# DETERMING THE ACCURACY VALUE OF THE TEXT FILES IN THE TWO FOLDERS
seq = SequenceMatcher(a=fin1, b=fin2)
print(seq.ratio())

# ACCURACY VALUE = 0.06164789567279194

0.30982905982905984


In [23]:
# GROUND TRUTHS VS TEXT FROM CONTOUR IMAGES

from difflib import SequenceMatcher

# PATH TO THE GROUND TRUTH TEXT FILES
path1 = glob.glob("/content/gdrive/MyDrive/CA_3/ground_truths/*.txt")

# LOOPING THROUGH THE FOLDER CONTAINING THE GROUND TRUTH TEXT FILES
for path1 in os.listdir('/content/gdrive/MyDrive/CA_3/ground_truths/'):
  with open('/content/gdrive/MyDrive/CA_3/ground_truths/'+ path1) as f1:
    fin1 = f1.read()

# PATH TO TEXT FILES FROM THE CONTOUR IMAGES
path2 = glob.glob("/content/gdrive/MyDrive/CA_3/Contour_text/*.txt")

# LOOPING THROUGH THE FOLDER CONTAINING THE CONTOUR TEXT FILES
for path2 in os.listdir('/content/gdrive/MyDrive/CA_3/Contour_text/'):
  with open('/content/gdrive/MyDrive/CA_3/Contour_text/'+ path2) as f2:
    fin2 = f2.read()

# DETERMING THE ACCURACY VALUE OF THE TEXT FILES IN THE TWO FOLDERS
seq = SequenceMatcher(a=fin1, b=fin2)
print(seq.ratio())

# ACCURACY VALUE = 0.30982905982905984

0.30982905982905984


In [24]:
# GROUND TRUTHS VS TEXT FROM EROSION IMAGES

from difflib import SequenceMatcher

# PATH TO THE GROUND TRUTH TEXT FILES
path1 = glob.glob("/content/gdrive/MyDrive/CA_3/ground_truths/*.txt")

# LOOPING THROUGH THE FOLDER CONTAINING THE GROUND TRUTH TEXT FILES
for path1 in os.listdir('/content/gdrive/MyDrive/CA_3/ground_truths/'):
  with open('/content/gdrive/MyDrive/CA_3/ground_truths/'+ path1) as f1:
    fin1 = f1.read()

# PATH TO TEXT FILES FROM THE EROSION IMAGES
path2 = glob.glob("/content/gdrive/MyDrive/CA_3/Erosion_text/*.txt")

# LOOPING THROUGH THE FOLDER CONTAINING THE EROSION TEXT FILES
for path2 in os.listdir('/content/gdrive/MyDrive/CA_3/Erosion_text/'):
  with open('/content/gdrive/MyDrive/CA_3/Erosion_text/'+ path2) as f2:
    fin2 = f2.read()

# DETERMING THE ACCURACY VALUE OF THE TEXT FILES IN THE TWO FOLDERS
seq = SequenceMatcher(a=fin1, b=fin2)
print(seq.ratio())

# ACCURACY VALUE = 0.3094983991462113

0.3094983991462113


In [25]:
# GROUND TRUTHS VS TEXT FROM GRAYSCALE IMAGES

from difflib import SequenceMatcher

# PATH TO THE GROUND TRUTH TEXT FILES
path1 = glob.glob("/content/gdrive/MyDrive/CA_3/ground_truths/*.txt")

# LOOPING THROUGH THE FOLDER CONTAINING THE GROUND TRUTH TEXT FILES
for path1 in os.listdir('/content/gdrive/MyDrive/CA_3/ground_truths/'):
  with open('/content/gdrive/MyDrive/CA_3/ground_truths/'+ path1) as f1:
    fin1 = f1.read()

# PATH TO TEXT FILES FROM THE GRAYSCALE IMAGES
path2 = glob.glob("/content/gdrive/MyDrive/CA_3/Grayscale_text/*.txt")

# LOOPING THROUGH THE FOLDER CONTAINING THE GRAYSCALE TEXT FILES
for path2 in os.listdir('/content/gdrive/MyDrive/CA_3/Grayscale_text/'):
  with open('/content/gdrive/MyDrive/CA_3/Grayscale_text/'+ path2) as f2:
    fin2 = f2.read()

# DETERMING THE ACCURACY VALUE OF THE TEXT FILES IN THE TWO FOLDERS
seq = SequenceMatcher(a=fin1, b=fin2)
print(seq.ratio())

# ACCURACY VALUE = 0.23939393939393938

0.23939393939393938


In [26]:
# GROUND TRUTHS VS TEXT FROM AVERAGE BLUR IMAGES

from difflib import SequenceMatcher

# PATH TO THE GROUND TRUTH TEXT FILES
path1 = glob.glob("/content/gdrive/MyDrive/CA_3/ground_truths/*.txt")

# LOOPING THROUGH THE FOLDER CONTAINING THE GROUND TRUTH TEXT FILES
for path1 in os.listdir('/content/gdrive/MyDrive/CA_3/ground_truths/'):
  with open('/content/gdrive/MyDrive/CA_3/ground_truths/'+ path1) as f1:
    fin1 = f1.read()

# PATH TO TEXT FILES FROM THE GRAYSCALE IMAGES
path2 = glob.glob("/content/gdrive/MyDrive/CA_3/Blur_text/*.txt")

# LOOPING THROUGH THE FOLDER CONTAINING THE GRAYSCALE TEXT FILES
for path2 in os.listdir('/content/gdrive/MyDrive/CA_3/Blur_text/'):
  with open('/content/gdrive/MyDrive/CA_3/Blur_text/'+ path2) as f2:
    fin2 = f2.read()

# DETERMING THE ACCURACY VALUE OF THE TEXT FILES IN THE TWO FOLDERS
seq = SequenceMatcher(a=fin1, b=fin2)
print(seq.ratio())

# ACCURACY VALUE = 0.27061310782241016

0.23939393939393938


In [27]:
# GROUND TRUTHS VS TEXT FROM GAUSSIAN BLUR IMAGES

from difflib import SequenceMatcher

# PATH TO THE GROUND TRUTH TEXT FILES
path1 = glob.glob("/content/gdrive/MyDrive/CA_3/ground_truths/*.txt")

# LOOPING THROUGH THE FOLDER CONTAINING THE GROUND TRUTH TEXT FILES
for path1 in os.listdir('/content/gdrive/MyDrive/CA_3/ground_truths/'):
  with open('/content/gdrive/MyDrive/CA_3/ground_truths/'+ path1) as f1:
    fin1 = f1.read()

# PATH TO TEXT FILES FROM THE GAUSSIAN BLUR IMAGES
path2 = glob.glob("/content/gdrive/MyDrive/CA_3/GaussianBlur_text/*.txt")

# LOOPING THROUGH THE FOLDER CONTAINING THE GAUSSIAN BLUR TEXT FILES
for path2 in os.listdir('/content/gdrive/MyDrive/CA_3/GaussianBlur_text/'):
  with open('/content/gdrive/MyDrive/CA_3/GaussianBlur_text/'+ path2) as f2:
    fin2 = f2.read()

# DETERMING THE ACCURACY VALUE OF THE TEXT FILES IN THE TWO FOLDERS
seq = SequenceMatcher(a=fin1, b=fin2)
print(seq.ratio())

# ACCURACY VALUE = 0.27061310782241016

0.23939393939393938


**ACCURACY VALUES**

The accuracy values were computed for each technique used and compared against the Ground Truth text files. The resultant images and text files gotten from each technique applied were saved in respective folders for clarity. The comparison of the techniques used against the Ground Truths were made folder-wise. The accuracy values are stated below.



1.   Ground Truth against Text from Original Images = 0.2485

2.   Ground Truth against Text from Region of Interest = 0.0616

3.   Ground Truth against Text from Contour Images = 0.3098

4.   Ground Truth against Text from Erosion Images = 0.3095

5.   Ground Truth against Text from Grayscale Images = 0.2394

6.   Ground Truth against Text from Average Blur Images = 0.2706

7.   Ground Truth against Text from Gaussian Blur Images = 0.2706

**CONCLUSION:** For this set of images, the Contour detection and Erosion techniques have the highest approximate accuracy values than the other techniques used.








